### PR TITLE
Bugfix/issue 6 : Trigger 'receive' callback

### PR DESCRIPTION
**Pull request by adynis**
Resolving issue #6 .
Now the 'receive' callback function from ESP32Ping.gpp (PingClass::_ping_recv_cb) is triggered !!

### CODE REVIEW

File: ping.cpp [hunk 1]
```diff
@@ -362,3 +369,4 @@ bool ping_regist_sent(struct ping_option *ping_opt, ping_sent_function ping_sent
     ping_opt->sent_function = ping_sent;
     return true;
 }
+*/
```
Comment (marian-craciunescu) [inline ping.cpp]: add new line at the file end

File: ping.h [hunk 2]
```diff
@@ -28,7 +28,7 @@ struct ping_resp {
 };
 
 bool ping_start(struct ping_option *ping_opt);
-void ping(const char *name, int count, int interval, int size, int timeout);
-bool ping_start(IPAddress adr, int count, int interval, int size, int timeout);
+//void ping(const char *name, int count, int interval, int size, int timeout);
+bool ping_start(struct ping_option *ping_o,IPAddress adr, int count, int interval, int size, int timeout);
```
Comment (marian-craciunescu) [inline ping.h]: this need to be refactored.

Comment (marian-craciunescu) [reply ping.h]: according with the previous pull request.

**Review by marian-craciunescu — CHANGES_REQUESTED** [review]

File: ping.h (hunk 2 above)
Comment (adynis) [reply ping.h]: Then I propose that I will change to something like: 
- Make the ping_option pointer as "optional" , something like: struct ping_option *ping_o = NULL , and inside the function we will compute all the "Statistics' and we will call the callback function only if we have a valid ping_o pointer. This way the ping() function will call ping_start() without ping_o pointer and PingClass::ping() can call it WITH ping_o parameter

What do you think ?

Comment (marian-craciunescu) [reply ping.h]: yes.I think this is the correct aproach here.

**Review by marian-craciunescu — APPROVED** [review]

**Pull request merged**